### PR TITLE
Task 4-5

### PR DESCRIPTION
**Pull request by Andrii-Kutil**

### CODE REVIEW

File: src/main/java/springboot/demo/controller/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import springboot.demo.model.dto.ProductDto;
+import springboot.demo.service.ProductService;
+
+@RestController
+public class ProductController {
+    @Autowired
+    ProductService productService;
+
+    @GetMapping("/most-commented-products")
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/controller/ProductController.java:18]: /products/most-commented
that's more logical

File: src/main/java/springboot/demo/controller/ReviewController.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@RestController
+public class ReviewController {
+
+    @Autowired
+    private SortWords sortWords;
+
+    @GetMapping("/most-popular-words")
+    private Map<String, Long> findMostPopularWords() {
+        return sortWords.findMostPopularWords();
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/controller/ReviewController.java:17]: same here, + why didn't you use same approach as with most commented products, where use can pass limit to controller?

File: src/main/java/springboot/demo/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import springboot.demo.model.dto.UserDto;
+import springboot.demo.service.UserService;
+
+@RestController
+public class UserController {
+    @Autowired
+    UserService userService;
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/controller/UserController.java:16]: access modifier?

File: src/main/java/springboot/demo/model/Product.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@Builder
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/model/Product.java:13]: do we need builder for model with one field?

File: src/main/java/springboot/demo/service/impl/ProductServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+import springboot.demo.model.Product;
+import springboot.demo.model.dto.ProductDto;
+import springboot.demo.repository.ProductRepository;
+import springboot.demo.service.ProductService;
+
+@Component
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/service/impl/ProductServiceImpl.java:12]: Why not @Service?

File: src/main/java/springboot/demo/service/impl/RoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package springboot.demo.service.impl;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import springboot.demo.model.Role;
+import springboot.demo.repository.RoleRepository;
+import springboot.demo.service.RoleService;
+
+@Component
```
Comment (RomanDubovskyi) [inline src/main/java/springboot/demo/service/impl/RoleServiceImpl.java:10]: same

File: src/main/java/springboot/demo/model/Product.java (hunk 4 above)
Comment (Andrii-Kutil) [reply src/main/java/springboot/demo/model/Product.java:13]: You are very attentive! The eagle eye)

File: src/main/java/springboot/demo/controller/ReviewController.java (hunk 2 above)
Comment (Andrii-Kutil) [reply src/main/java/springboot/demo/controller/ReviewController.java]: ok, I will use pagination.

**Pull request merged**